### ISSUE TITLE
[Py] Retrieve External Pinned Maps

### ISSUE THREAD

**Issue by gamemann** [iovisor/bcc#3517]
Hi everyone,

I was wondering if BCC's Python library supports retrieving a BPF map from the file system (AKA a pinned map). I am loading an XDP program in C using [libbpf](https://github.com/libbpf/libbpf/) and pinning maps to the file system. 

I am using a higher-level library in Python to retrieve information and trying to update the BPF maps with this information.

I've searched the reference guide [here](https://github.com/iovisor/bcc/blob/master/docs/reference_guide.md#bcc-python) and wasn't able to find any way to do this. It appears to be possible in BCC C according to [here](https://github.com/iovisor/bcc/blob/master/docs/reference_guide.md#pinned-maps), though. I also believe this is similar to the following.

https://github.com/iovisor/bcc/issues/2223

If this is currently not supported, is there any plans to add support in the future?

Any help is highly appreciated and thank you!

**Comment by yonghong-song** [comment]
I think it should work for python. You can use `BPF_TABLE_PINNED` in the bpf program and then use bpf.get_table(table_name) in your python code to get the map and from there you can access map elements.
Could you give a try?

**Comment by gamemann** [comment]
@yonghong-song Thank you for your response! To my understanding, this means I'd need to load the BPF program inside of the Python program. I was wondering if there's a way to retrieve the BPF map pinned on the file system by another BPF program using BCC in Python.

According to the BPF constructor [here](https://github.com/iovisor/bcc/blob/master/docs/reference_guide.md#1-bpf), I'd need to load a BPF program and then use `BPF.get_table(name)`.

Thank you!

**Comment by jschwinger233** [comment]
> According to the BPF constructor here, I'd need to load a BPF program and then use BPF.get_table(name).

That's true, but we can use a simplified BPF program to instantiate the `bcc.BPF()`. 

For example, if we defined a pinned map in bpf program:
```c
BPF_TABLE_PINNED("percpu_hash", u32, struct datarec, xdp_stats_map, XDP_ACTION_MAX, "/sys/fs/bpf/xdp_stats_map");
```
then in order to retrieve the map items in an independent BCC python program, all we need to feed the `bcc.BPF()` is this piece of code to let it know where to find the pinned map and how the kv types are defined.
```python
xdp_stats_map = bcc.BPF(text=r'BPF_TABLE_PINNED("percpu_hash", u32, struct datarec, xdp_stats_map, XDP_ACTION_MAX, "/sys/fs/bpf/xdp_stats_map");').get_table('xdp_stats_map')
```
